arch/riscv/core/smp.c

```diff
@@ -74,11 +75,15 @@ void arch_secondary_cpu_init(int hartid)
 #endif
 #ifdef CONFIG_SMP
 	irq_enable(RISCV_IRQ_MSOFT);
+#endif
+#ifdef CONFIG_RISCV_SOC_HAS_CUSTOM_PER_CORE_INIT
+	z_soc_per_core_init();
```

Comment:
I think there should be an entry for this in `arch_kernel_init()` similar to the `z_riscv_pmp_init()`, so that the function is ran by the main core as well - https://github.com/zephyrproject-rtos/zephyr/blob/main/arch/riscv/include/kernel_arch_func.h#L54

Reply:
This probably should be in a separate commit as well